### PR TITLE
HRT-65 Added sign up to use_core_auth_api

### PR DESCRIPTION
**Pull request by Dizzzmas**

### CODE REVIEW

File: jetkit/api/auth/__init__.py [hunk 1]
```diff
@@ -66,6 +69,24 @@ def login(email: str, password: str):
...
+    @blp.route("sign-up", methods=["POST"])
+    @blp.response(user_schema)
+    @blp.arguments(LoginRequest, as_kwargs=True)
+    def sign_up(email: str, password: str):
+        """Sign up with email and password. Possibly add other fields later."""
+        cleaned_email = email.strip().lower()
+        existing_user: AuthModel = auth_model.query.filter_by(email=cleaned_email).one_or_none()
+        if existing_user:
+            abort(400, message="There's already a registered user with this email")
+        values = dict(email=email, _password=generate_password_hash(password), _user_type=CoreUserType.normal.value)
```
Comment (Dizzzmas) [inline jetkit/api/auth/__init__.py]: IDK, is there currently in jetkit a better way to insert a user?

Comment (revmischa) [reply jetkit/api/auth/__init__.py]: There is really no reason to use upsert here at all.
what's wrong with:
```python
user = User(email=email, password=password)
session = auth_model.query.session
session.add(user)
session.commit()
```

File: jetkit/api/auth/__init__.py [hunk 2]
```diff
@@ -66,6 +69,24 @@ def login(email: str, password: str):
             abort(401, message="Wrong user name or password")
         return auth_response_for_user(user)
 
+    @blp.route("sign-up", methods=["POST"])
+    @blp.response(user_schema)
+    @blp.arguments(LoginRequest, as_kwargs=True)
+    def sign_up(email: str, password: str):
+        """Sign up with email and password. Possibly add other fields later."""
+        cleaned_email = email.strip().lower()
```
Comment (rkyslyy) [inline jetkit/api/auth/__init__.py]: Email usernames (unlike email domain names) are case sensitive. If we do `.lower()` here, it won't be possible to register users with emails `foobar@gmail.com` and `fooBar@gmail.com`, which are totally different

Comment (revmischa) [reply jetkit/api/auth/__init__.py]: Roman, can you cite your source here? I've never heard of that. You're saying I could register Rkyslyy@gmail.com even if you have rkyslyy@gmail.com?
I'm almost certain it's correct to lowercase email addresses before storing them

Comment (rkyslyy) [reply jetkit/api/auth/__init__.py]: Huh, sorry my bad
Looking at other sources and trying to register Rkyslyy@gmail.com I see that it's the same thing

Please disregard the comment above

**Pull request merged**